api/httpclient.go

```diff
@@ -0,0 +1,140 @@
+package api
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"strings"
+	"time"
+)
+
+const (
+	// Default http client timeout in secs.
+	defaultHttpClientTimeout = 10 * time.Second
+)
+
+type (
+	// Client is the base for http/https calls
+	Client struct {
+		httpClient    *http.Client
+		Debug         bool
+		BaseUrl       string
+		RequestFilter func(info RequestInfo) (req *http.Request, err error)
+	}
+
+	// ClientConf Models http client configurations
+	ClientConf struct {
+		Debug   bool
+		BaseUrl string
+	}
+
+	// RequestInfo models the http request data.
+	RequestInfo struct {
+		client  *Client
+		request *http.Request
+		Payload interface{}
+		Method  string
+		Url     string
+	}
+)
+
+// NewClient return a new HTTP client
+func NewClient(conf *ClientConf) (c *Client) {
+	t := http.DefaultTransport.(*http.Transport).Clone()
+	t.TLSClientConfig = &tls.Config{
+		InsecureSkipVerify: true,
```

Comment:
I'm not familiar with TLS in Go, but if it says "disable certificate checks", perhaps it's better to enable it by default to be safe, and explicitly disable in test code?